### ISSUE TITLE
bundle creation fails on macOS Sonoma with qt6

### ISSUE THREAD

**Issue by Mattoje** [kvirc/KVIrc#2602]
Latest git, macOS 14.3.1, xcode 14.2, qt 6.6.1 (homebrew)

Everything works fine with qt5 but with qt6 fixup_bundle fails (after a bunch of warnings) with:

```
-- fixup_bundle
--   app='/Users/matteo/ports/KVIrc/build/KVIrc.app'
--   libs=''
--   dirs='/'
--   ignoreItems='Python;libqsqlodbc.dylib;libqsqlpsql.dylib'
-- fixup_bundle: preparing...
-- 
warning: cannot resolve item '@rpath/QtQuickWidgets.framework/Versions/A/QtQuickWidgets'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQuickWidgets.framework/Versions/A/QtQuickWidgets' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQuickWidgets.framework/Versions/A/QtQuickWidgets'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtPrintSupport.framework/Versions/A/QtPrintSupport'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtPrintSupport.framework/Versions/A/QtPrintSupport' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtPrintSupport.framework/Versions/A/QtPrintSupport'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtWebEngineCore.framework/Versions/A/QtWebEngineCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtWebEngineCore.framework/Versions/A/QtWebEngineCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtWebEngineCore.framework/Versions/A/QtWebEngineCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtWebChannel.framework/Versions/A/QtWebChannel'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtWebChannel.framework/Versions/A/QtWebChannel' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtWebChannel.framework/Versions/A/QtWebChannel'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtPositioning.framework/Versions/A/QtPositioning'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtPositioning.framework/Versions/A/QtPositioning' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtPositioning.framework/Versions/A/QtPositioning'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtQuick.framework/Versions/A/QtQuick'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQuick.framework/Versions/A/QtQuick' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQuick.framework/Versions/A/QtQuick'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtQml.framework/Versions/A/QtQml'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQml.framework/Versions/A/QtQml' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQml.framework/Versions/A/QtQml'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtWidgets.framework/Versions/A/QtWidgets'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtWidgets.framework/Versions/A/QtWidgets' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtWidgets.framework/Versions/A/QtWidgets'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtWidgets.framework/Versions/A/QtWidgets'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtWidgets.framework/Versions/A/QtWidgets' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtWebChannel.framework/Versions/A/QtWebChannel'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtWebChannel.framework/Versions/A/QtWebChannel' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtPositioning.framework/Versions/A/QtPositioning'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtPositioning.framework/Versions/A/QtPositioning' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQuick.framework/Versions/A/QtQuick'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQuick.framework/Versions/A/QtQuick' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQml.framework/Versions/A/QtQml'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQml.framework/Versions/A/QtQml' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQml.framework/Versions/A/QtQml'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQml.framework/Versions/A/QtQml' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQmlModels.framework/Versions/A/QtQmlModels' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQml.framework/Versions/A/QtQml'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQml.framework/Versions/A/QtQml' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtOpenGL.framework/Versions/A/QtOpenGL' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtQml.framework/Versions/A/QtQml'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtQml.framework/Versions/A/QtQml' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtNetwork.framework/Versions/A/QtNetwork'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtNetwork.framework/Versions/A/QtNetwork' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtGui.framework/Versions/A/QtGui'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtGui.framework/Versions/A/QtGui' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtDBus.framework/Versions/A/QtDBus'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtDBus.framework/Versions/A/QtDBus' returning type 'other' -- possibly incorrect
-- 
warning: cannot resolve item '@rpath/QtDBus.framework/Versions/A/QtDBus'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtCore.framework/Versions/A/QtCore'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- warning: gp_resolved_file_type non-absolute file '@rpath/QtCore.framework/Versions/A/QtCore' returning type 'other' -- possibly incorrect
-- warning: embedded item does not exist '/usr/opt/libpng/lib/libpng16.16.dylib'
-- 
warning: cannot resolve item '@loader_path/../../../../opt/libpng/lib/libpng16.16.dylib'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

-- 
warning: cannot resolve item '@rpath/QtDBus.framework/Versions/A/QtDBus'

  possible problems:
    need more directories?
    need to use InstallRequiredSystemLibraries?
    run in install tree instead of build tree?

CMake Error at /Applications/CMake.app/Contents/share/cmake-3.28/Modules/BundleUtilities.cmake:458 (message):
  otool -l failed: 1

  error:
  /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/otool-classic:
  can't open file: @rpath/QtDBus.framework/Versions/A/QtDBus (No such file or
  directory)

Call Stack (most recent call first):
  /Applications/CMake.app/Contents/share/cmake-3.28/Modules/BundleUtilities.cmake:527 (get_item_rpaths)
  /Applications/CMake.app/Contents/share/cmake-3.28/Modules/BundleUtilities.cmake:649 (set_bundle_key_values)
  /Applications/CMake.app/Contents/share/cmake-3.28/Modules/BundleUtilities.cmake:933 (get_bundle_keys)
  src/kvirc/cmake_install.cmake:87 (fixup_bundle)
  src/cmake_install.cmake:47 (include)
  cmake_install.cmake:62 (include)


make: *** [install] Error 1
```

**Comment by ctrlaltca** [comment]
Mac and Windows installer have not been ported to Qt6 yet.
They may work with some small modifications, but nobody tested them out.

**Comment by Mattoje** [comment]
i'll be happy to  beta test if needed

**Comment by ctrlaltca** [comment]
@Mattoje : https://github.com/kvirc/KVIrc/pull/2612 switches the mac build to Qt6.
A prebuilt DMG is available at https://github.com/kvirc/KVIrc/actions/runs/8129358172?pr=2612 , can you please test if it works for you?

**Comment by Mattoje** [comment]
Unfortunately it does not work, but i will try to compile it on my own.


Anyway here is the crash report:

Process:               kvirc [17135]
Path:                  /Volumes/VOLUME/*/KVIrc.app/Contents/MacOS/kvirc
Identifier:            com.kvirc.kvirc
Version:               5.2.0 (???)
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               502

Date/Time:             2024-03-03 19:18:24.4011 +0100
OS Version:            macOS 14.3.1 (23D60)
Report Version:        12
Anonymous UUID:        7F606798-367A-0A07-9115-71CE579F49A9

Sleep/Wake UUID:       8942B0E3-4511-4400-B565-8A40DED6112F

Time Awake Since Boot: 22000 seconds
Time Since Wake:       1175 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: @rpath/libkvilib.5.dylib
Referenced from: <D0B705A7-1087-32DB-9030-65426831F064> /Volumes/VOLUME/*/KVIrc.app/Contents/MacOS/kvirc
Reason: tried: '/usr/local/lib/libkvilib.5.dylib' (no such file), '/usr/lib/libkvilib.5.dylib' (no such file, not in dyld cache)
(terminated at launch; ignore backtrace)

**Comment by Mattoje** [comment]
Ok i tried building my own and i managed to get it running with some trickery.
Here is what i did:

1. after make install i had to manually run macdeployqt "build/KVIrc.app" -libpath=build/KVIrc.app/Contents/MacOS
2. cd build/KVIrc.app/Contents/MacOS and run:
     install_name_tool -change @rpath/libkvilib.5.dylib @executable_path/libkvilib.5.dylib kvirc
3. still in build/KVIrc.app/Contents/MacOS run:
     1. otool -L kvirc |grep rpath |cut -c 9- |awk '{print "install_name_tool -change @rpath/"$1,"@executable_path/../Frameworks/"$1" kvirc"}' >fix.sh
     2. sh fix.sh

After all this mess KVirc works.

There must be a cmake way to do the install_name_tool thing, i'm still investigating

**Comment by ctrlaltca** [comment]
@Mattoje  Thank you for investigating the problem.
So the files are in place, but the binary doesn't report the correct paths.
I've tried to move kvilib and modules into Frameworks, hoping that this will make macdeployqt able to find them and fix the paths automatically.
The build is ongoing at https://github.com/kvirc/KVIrc/actions/runs/8139988384?pr=2612
Unfortunately i don't have a mac here so i can't test if the solution works, so i would be glad if you give it a try 

EDIT: looking at the build output it probably didn't work :(

**Comment by Mattoje** [comment]
i finally found a "works for me" fix:

[works.for.me.patch](https://github.com/kvirc/KVIrc/files/14500189/works.for.me.patch)

kudos to:
https://root-forum.cern.ch/t/rpath-setting-with-cmake-for-locally-installed-root-conda-macos/53983
https://root-forum.cern.ch/t/rpath-mac-os-x-and-how-to-load-a-library-from-anywhere-and-execute-macros-from-anywhere/49558/3
P.S. dont' forget to add: macdeployqt KVIrc.app -appstore-compliant after make install to https://github.com/kvirc/KVIrc/wiki/Compiling-KVIrc-on-macOS

**Comment by ctrlaltca** [comment]
@Mattoje wonderful! Added your patch, it's currently building at https://github.com/kvirc/KVIrc/actions/runs/8162781648?pr=2612

**Comment by ctrlaltca** [comment]
Mattoje: can you please confirm the build at https://github.com/kvirc/KVIrc/actions/runs/8162781648?pr=2612 works for you? If it does I'm going to merge #2612

**Comment by Mattoje** [comment]
Sadly it does not with this python related error:

Library not loaded: /Library/Frameworks/Python.framework/Versions/3.11/Python

This is kinda weird because the "bundling process" should have taken care of putting Python Framework into the bundle itself.

By default macos (at least mine) does not ship with Python Framework installed

Maybe a partial solution is to disable python support in KVirc
Anyway the full dump:

Process:               kvirc [1237]
Path:                  /Volumes/VOLUME/*/KVIrc.app/Contents/MacOS/kvirc
Identifier:            com.kvirc.kvirc
Version:               5.2.0 (???)
Code Type:             X86-64 (Native)
Parent Process:        launchd [1]
User ID:               502

Date/Time:             2024-03-07 11:15:24.8795 +0100
OS Version:            macOS 14.3.1 (23D60)
Report Version:        12
Anonymous UUID:        7F606798-367A-0A07-9115-71CE579F49A9


Time Awake Since Boot: 290 seconds

System Integrity Protection: enabled

Crashed Thread:        0

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000

Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: /Library/Frameworks/Python.framework/Versions/3.11/Python
Referenced from: <7EBC29FB-CBB1-327C-89E0-5141782CABDC> /Volumes/VOLUME/*/KVIrc.app/Contents/MacOS/kvirc
Reason: tried: '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file), '/System/Library/Frameworks/Python.framework/Versions/3.11/Python' (no such file, not in dyld cache)
(terminated at launch; ignore backtrace)

**Comment by ctrlaltca** [comment]
Fixed by #2612
